test/checks/label/label-content-name-mismatch.js

```diff
@@ -0,0 +1,127 @@
+		var node = fixture.querySelector('#target');
+		var tree = (axe._tree = axe.utils.getFlattenedTree(fixture));
+		var actual = check.evaluate(
+			node,
+			options,
+			axe.utils.getNodeFromTree(tree[0], node)
+		);
+		assert.isFalse(actual);
+	});
+
+	it('returns false when accessible text does not contain text content', function() {
+		fixture.innerHTML =
+			'<button id="target" name="link" aria-label="Ok">test page</button>';
+		var node = fixture.querySelector('#target');
+		var tree = (axe._tree = axe.utils.getFlattenedTree(fixture));
+		var actual = check.evaluate(
+			node,
+			options,
+			axe.utils.getNodeFromTree(tree[0], node)
+		);
+		assert.isFalse(actual);
+	});
+
+	it('returns true when visible label and accessible name matches when trailing white spaces are removed', function() {
+		fixture.innerHTML =
+			'<div id="target" role="link" aria-label="next page ">next page</div>';
+		var node = fixture.querySelector('#target');
+		var tree = (axe._tree = axe.utils.getFlattenedTree(fixture));
+		var actual = check.evaluate(
+			node,
+			options,
+			axe.utils.getNodeFromTree(tree[0], node)
+		);
+		assert.isTrue(actual);
+	});
+
+	it('returns true when visible label and accessible name matches after handling character insensitivity', function() {
+		fixture.innerHTML =
+			'<div id="target" role="link" aria-label="Next Page">next page</div>';
+		var node = fixture.querySelector('#target');
+		var tree = (axe._tree = axe.utils.getFlattenedTree(fixture));
+		var actual = check.evaluate(
+			node,
+			options,
+			axe.utils.getNodeFromTree(tree[0], node)
+		);
+		assert.isTrue(actual);
+	});
+
+	it('returns true when full visible label is contained in the accessible name', function() {
+		fixture.innerHTML =
+			'<button id="target" name="link" aria-label="Next Page in the list">Next Page</button>';
+		var node = fixture.querySelector('#target');
+		var tree = (axe._tree = axe.utils.getFlattenedTree(fixture));
+		var actual = check.evaluate(
+			node,
+			options,
+			axe.utils.getNodeFromTree(tree[0], node)
+		);
+		assert.isTrue(actual);
+	});
+
+	it('returns false when visible label doesn’t match accessible name', function() {
+		fixture.innerHTML =
+			'<div id="target" role="link" aria-label="OK">Next</div>';
+		var node = fixture.querySelector('#target');
+		var tree = (axe._tree = axe.utils.getFlattenedTree(fixture));
+		var vNode = axe.utils.getNodeFromTree(tree[0], node);
+		var actual = check.evaluate(node, options, vNode);
+		assert.isFalse(actual);
+	});
+
+	it('returns false when not all of visible label is included in accessible name', function() {
+		fixture.innerHTML =
+			'<button id="target" name="link" aria-label="the full">The full label</button>';
+		var node = fixture.querySelector('#target');
+		var tree = (axe._tree = axe.utils.getFlattenedTree(fixture));
+		var vNode = axe.utils.getNodeFromTree(tree[0], node);
+		var actual = check.evaluate(node, options, vNode);
+		assert.isFalse(actual);
+	});
+});
```

Comment:
You're missing quite a few tests here:
- aria-labelledby with reference to an element that matches / that doesn't match
- aria-labelledby with references to multiple elements
- aria-labelledby with references to non-existing elements
- Elements with hidden texts like: `<button aria-label="help">? <i class="sr-only">help</i></button>
- Elements with ASCII as non-text content `<button aria-label="close">X</button>`
- ...